masonry/src/widgets/slider.rs

```diff
@@ -0,0 +1,393 @@
+
+    /// Sets the track color of the slider.
+    pub fn set_track_color(this: &mut WidgetMut<'_, Self>, color: Option<Color>) {
+        if this.widget.track_color != color {
+            this.widget.track_color = color;
+            this.ctx.request_render();
+        }
+    }
+
+    /// Sets the active track color of the slider.
+    pub fn set_active_track_color(this: &mut WidgetMut<'_, Self>, color: Option<Color>) {
+        if this.widget.active_track_color != color {
+            this.widget.active_track_color = color;
+            this.ctx.request_render();
+        }
+    }
+    /// Sets the thumb color of the slider.
+    pub fn set_thumb_color(this: &mut WidgetMut<'_, Self>, color: Option<Color>) {
+        if this.widget.thumb_color != color {
+            this.widget.thumb_color = color;
+            this.ctx.request_render();
+        }
+    }
+    /// Sets the track thickness of the slider.
+    pub fn set_track_thickness(this: &mut WidgetMut<'_, Self>, thickness: Option<f64>) {
+        if this.widget.track_thickness != thickness {
+            this.widget.track_thickness = thickness;
+            this.ctx.request_layout(); // El tamaño cambia, pedimos layout
+        }
+    }
+    /// Sets the thumb radius of the slider.
+    pub fn set_thumb_radius(this: &mut WidgetMut<'_, Self>, radius: Option<f64>) {
+        if this.widget.thumb_radius != radius {
+            this.widget.thumb_radius = radius;
+            this.ctx.request_layout(); // El tamaño cambia, pedimos layout
+        }
+    }
+
+    // --- Logic ---
+    fn update_value_from_position(&mut self, x: f64, width: f64) -> bool {
+        let base_thumb_radius = self.thumb_radius.unwrap_or(6.0);
+        let thumb_radius = if self.is_dragging {
+            base_thumb_radius + 2.0
+        } else {
+            base_thumb_radius
+        };
+        let track_start_x = thumb_radius;
+        let track_width = width - (thumb_radius * 2.0);
+        let relative_x = x - track_start_x;
+
+        let progress = (relative_x / track_width).clamp(0.0, 1.0);
+        let new_value = self.min + progress * (self.max - self.min);
+        let old_value = self.value;
+
+        let final_value = if let Some(step) = self.step {
+            ((new_value / step).round() * step).clamp(self.min, self.max)
+        } else {
+            new_value.clamp(self.min, self.max)
+        };
+
+        if (final_value - old_value).abs() > f64::EPSILON {
+            self.value = final_value;
+            true
+        } else {
+            false
+        }
+    }
+}
+
+impl Widget for Slider {
+    type Action = f64;
+
+    fn accepts_pointer_interaction(&self) -> bool {
+        !self.disabled
+    }
+
+    fn on_pointer_event(
+        &mut self,
+        ctx: &mut EventCtx<'_>,
+        _props: &mut PropertiesMut<'_>,
+        event: &PointerEvent,
+    ) {
+        if self.disabled {
+            return;
+        }
+        match event {
+            PointerEvent::Down { state, .. } => {
```

Comment:
@xorgy Can you take a look at this section? I feel like something needs to change here ...

Reply:
I think the issue is that this will accept any pointer button as well as touch, which is probably not correct. You could look at other instances where we match pointer down for dragging in Masonry for some references.